grpc-sys/build.rs

```diff
@@ -12,146 +12,148 @@
-                "bench" | "release" => {
-                    zlib = "zlibstatic";
-                    "Release"
-                }
-                _ => {
-                    zlib = "zlibstaticd";
-                    "Debug"
-                }
-            };
-            println!("cargo:rustc-link-search=native={}/{}", build_dir, profile);
-            println!(
-                "cargo:rustc-link-search=native={}/third_party/cares/{}",
-                build_dir,
-                profile
-            );
-            println!(
-                "cargo:rustc-link-search=native={}/third_party/zlib/{}",
-                build_dir,
-                profile
-            );
-            println!(
-                "cargo:rustc-link-search=native={}/third_party/boringssl/ssl/{}",
-                build_dir,
-                profile
-            );
-            println!(
-                "cargo:rustc-link-search=native={}/third_party/boringssl/crypto/{}",
-                build_dir,
-                profile
-            );
-        } else {
-            println!("cargo:rustc-link-search=native={}", build_dir);
-            println!(
-                "cargo:rustc-link-search=native={}/third_party/cares",
-                build_dir
-            );
-            println!(
-                "cargo:rustc-link-search=native={}/third_party/zlib",
-                build_dir
-            );
-            println!(
-                "cargo:rustc-link-search=native={}/third_party/boringssl/ssl",
-                build_dir
-            );
-            println!(
-                "cargo:rustc-link-search=native={}/third_party/boringssl/crypto",
-                build_dir
-            );
-        }
+    println!("cargo:rustc-link-lib=static={}", zlib);
+    println!("cargo:rustc-link-lib=static=cares");
+    println!("cargo:rustc-link-lib=static=gpr");
+    println!("cargo:rustc-link-lib=static=grpc");
+    println!("cargo:rustc-link-lib=static=ssl");
+    println!("cargo:rustc-link-lib=static=crypto");
 
-        println!("cargo:rustc-link-lib=static={}", zlib);
-        println!("cargo:rustc-link-lib=static=cares");
-        println!("cargo:rustc-link-lib=static=gpr");
-        println!("cargo:rustc-link-lib=static=grpc");
-        println!("cargo:rustc-link-lib=static=ssl");
-        println!("cargo:rustc-link-lib=static=crypto");
+    cc.include("grpc/include");
+}
 
-        cc.include("grpc/include");
+fn get_env(name: &str) -> Option<String> {
+    println!("cargo:rerun-if-env-changed={}", name);
+    match env::var(name) {
+        Ok(s) => Some(s),
+        Err(VarError::NotPresent) => None,
+        Err(VarError::NotUnicode(s)) => {
+            panic!("unrecognize env var of {}: {:?}", name, s.to_string_lossy());
+        }
     }
 }
 
 fn main() {
     let mut cc = Build::new();
 
-    imp::build_or_link_grpc(&mut cc);
+    if get_env("GRPCIO_SYS_USE_PKG_CONFIG").map_or(false, |s| s == "1") {
```

Comment:
seem using 1 is to build grpc?

Reply:
`GRPCIO_SYS_USE_PKG_CONFIG=1` means use `pkg-config` to search for pre-installed grpc package and link to it.